misc/snapshotter/snapshotter.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# Copyright (c) 2023. Nydus Developers. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# Common nydus snapshotter config options
+FS_DRIVER="${FS_DRIVER:-fusedev}"
+LOG_LEVEL="${LOG_LEVEL:-info}"
+NYDUSD_DAEMON_MODE="${NYDUSD_DAEMON_MODE:-multiple}"
+
+# Options for blockdev
+ENABLE_KATA_VOLUME="${ENABLE_KATA_VOLUME:-true}"
+ENABLE_TARFS="${ENABLE_TARFS:-true}"
+MOUNT_TARFS_ON_HOST="${MOUNT_TARFS_ON_HOST:-false}"
+EXPORT_MODE="${EXPORT_MODE:-image_block_with_verity}"
+
+# The directory about nydus and nydus snapshotter
+NYDUS_CONFIG_DIR="${NYDUS_CONFIG_DIR:-/etc/nydus}"
+NYDUS_LIB_DIR="${NYDUS_LIB_DIR:-/var/lib/containerd-nydus}"
+NYDUS_RUN_DIR="${NYDUS_RUN_DIR:-/run/containerd-nydus}"
+NYDUS_BINARY_DIR="${NYDUS_BINARY_DIR:-/usr/local/bin}"
+SNAPSHOTTER_SCRYPT_DIR="${SNAPSHOTTER_SCRYPT_DIR:-/opt/nydus}"
+
+# The binary about nydusd, nydus-image and nydus-snapshotter
+NYDUSD_BINARY="${NYDUSD_BINARY:-${NYDUS_BINARY_DIR}/nydusd}"
+NYDUSIMAGE_BINARY="${NYDUSIMAGE_BINARY:-${NYDUS_BINARY_DIR}/nydus-image}"
+SNAPSHOTTER_BINARY="${SNAPSHOTTER_BINARY:-${NYDUS_BINARY_DIR}/containerd-nydus-grpc}"
+
+# The config about nydusd and nydus snapshotter
+NYDUSD_CONFIG="${NYDUSD_CONFIG:-${NYDUS_CONFIG_DIR}/nydusd-fusedev.json}"
+SNAPSHOTTER_CONFIG="${SNAPSHOTTER_CONFIG:-${NYDUS_CONFIG_DIR}/config-fusedev.toml}"
+
+# If true, the script would read the config from env.
+ENABLE_SNAPSHOTTER_CONFIG_FROM_VOLUME="${ENABLE_SNAPSHOTTER_CONFIG_FROM_VOLUME:-false}"
+
+COMMANDLINE=""
+
+# If we fail for any reason a message will be displayed
+die() {
+    msg="$*"
+    echo "ERROR: $msg" >&2
+    exit 1
+}
+
+function fs_driver_handler() {
+    case "${FS_DRIVER}" in
+    fusedev) SNAPSHOTTER_CONFIG="${NYDUS_CONFIG_DIR}/config-fusedev.toml" ;;
+    fscache) SNAPSHOTTER_CONFIG="${NYDUS_CONFIG_DIR}/config-fscache.toml" ;;
+    blockdev) SNAPSHOTTER_CONFIG="${NYDUS_CONFIG_DIR}/config-blockdev.toml" ;;
+    proxy) SNAPSHOTTER_CONFIG="${NYDUS_CONFIG_DIR}/config-proxy.toml" ;;
+    *) die "invalid fs driver" ;;
```

Comment:
I suggest to log the name of the invalid driver, e.g., `invalid fs driver: ${FS_DRIVER}`

Reply:
Fixed.